### PR TITLE
OCD-4377 - Add ability to store SSO user id when auditing changes

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/lastmodifieduserstrategy/LastModifiedUserStrategy.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.ff4j.FF4j;
+
+import gov.healthit.chpl.SpringContext;
+import gov.healthit.chpl.entity.EntityAudit;
+
+public abstract class LastModifiedUserStrategy {
+    private FF4j ff4j = SpringContext.getBean(FF4j.class);
+
+    public abstract void populationLastModifiedUser(EntityAudit entityAudit);
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/lastmodifieduserstrategy/LastModifiedUserStrategy.java]: Can you change to `populateLastModifiedUser` ?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/attestation/entity/AttestationPeriodDeveloperExceptionEntity.java [hunk 2]
```diff
@@ -13,23 +12,26 @@
...
 @Setter
 @ToString
-@Builder
+@SuperBuilder
 @AllArgsConstructor
 @NoArgsConstructor
 @Entity
 @Table(name = "attestation_period_developer_exception")
-public class AttestationPeriodDeveloperExceptionEntity {
+public class AttestationPeriodDeveloperExceptionEntity extends EntityAudit{
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/attestation/entity/AttestationPeriodDeveloperExceptionEntity.java]: checkstyle thing

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/attestation/entity/AttestationSubmissionEntity.java [hunk 3]
```diff
@@ -15,23 +14,26 @@
...
 @ToString
-@Builder
-@NoArgsConstructor
+@SuperBuilder
 @AllArgsConstructor
-public class AttestationSubmissionEntity {
+@NoArgsConstructor
+@Entity
+@Table(name = "attestation_submission")
+public class AttestationSubmissionEntity extends EntityAudit{
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/attestation/entity/AttestationSubmissionEntity.java]: checkstyle

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertificationIdDAO.java [hunk 4]
```diff
@@ -60,9 +59,6 @@ public CertificationIdDTO create(List<CertifiedProductDetailsDTO> listings, Stri
         entity = new CertificationIdEntity();
         entity.setCertificationId(this.generateCertificationIdString(listings, year));
         entity.setYear(year);
-        entity.setLastModifiedDate(new Date());
-        entity.setCreationDate(new Date());
-        entity.setLastModifiedUser(MODIFIED_USER_ID);
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertificationIdDAO.java:65]: Is MODIFIED_USER_ID still used?

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**